src/consumers/index.js

```diff
@@ -1,109 +1,42 @@
 const config = require('~/src/config')
-const amqplib = require('amqplib')
 const logger = require('~/src/logging').logger(module)
-
-const QueueConsumer = require('./QueueConsumer')
+const queue = require('windbreaker-service-util/queue')
 
 const EVENTS_QUEUE_NAME = config.getEventsQueueName()
 const EVENTS_QUEUE_PREFETCH_COUNT = config.getEventsQueuePrefetchCount()
 const WORK_QUEUE_NAME = config.getWorkQueueName()
 const WORK_QUEUE_PREFETCH_COUNT = config.getWorkQueuePrefetchCount()
-
 const CONSUMER_RECONNECT_TIMEOUT = config.getConsumerReconnectTimeout()
 
-let connectionPromise
-let amqConnection
-
-/**
- * Creates a connection
- */
-async function _getConnection () {
-  if (connectionPromise) {
-    return connectionPromise
-  }
-
-  if (!amqConnection) {
-    logger.info(`Attempting to connect to amq ${config.getAmqUrl()}`)
-    connectionPromise = amqplib.connect(config.getAmqUrl())
-      .then((connection) => {
-        amqConnection = connection
-        connectionPromise = null
-        return connection
-      })
-      .catch((err) => {
-        connectionPromise = null
-        amqConnection = null
-        throw new Error('Unable to connect to ActiveMQ', err)
-      })
-
-    return connectionPromise
-  } else {
-    return amqConnection
-  }
-}
-
-async function _startConsumer (queueName, prefetchCount) {
-  try {
-    const connection = await _getConnection()
-    const consumer = new QueueConsumer({
-      connection,
-      queueName,
-      prefetchCount
-    })
-
-    consumer.on('message', async (message) => {
-      try {
-        // TODO: handle messages here
-
-        // acknowledge the message on success
-        await consumer.acknowledgeMessage(message)
-      } catch (err) {
-        logger.error('Error handling message:', err)
-
-        // reject/requeue message
-        await consumer.rejectMessage(message)
-      }
-    })
-
-    const _closeConsumer = async () => {
-      await consumer.stop()
-      _restartConsumer(queueName, prefetchCount)
+exports.initialize = async function () {
+  const amqUrl = config.getAmqUrl()
+
+  const eventsConsumer = await queue.createConsumer({
+    amqUrl,
+    logger,
+    reconnectTimeout: CONSUMER_RECONNECT_TIMEOUT,
+    consumerOptions: {
+      queueName: EVENTS_QUEUE_NAME,
+      prefetchCount: EVENTS_QUEUE_PREFETCH_COUNT
+    },
+    async onMessage (message) {
+      // TODO: Handle the message
+      await Promise.resolve()
     }
-
-    const _handleConnectionClosed = (err) => {
-      logger.error('Connection error', err)
-      amqConnection = null
-      connectionPromise = null
-      _closeConsumer()
+  })
+
+  await queue.createConsumer({
+    amqUrl,
+    logger,
+    reconnectTimeout: CONSUMER_RECONNECT_TIMEOUT,
+    consumerOptions: {
+      queueName: WORK_QUEUE_NAME,
+      prefetchCount: WORK_QUEUE_PREFETCH_COUNT,
+      connection: eventsConsumer.getConnection()
```

Comment:
It looks like we do some connection sharing in the beginning, however in https://github.com/windbreaker-io/windbreaker-service-util/pull/3/files it looks like when a connection is lost or a consumer errors out, a new individual connection is being created when restarting the consumer. 

Connections seem relatively cheap and from some quick googling, `rabbitmq` seems to be handle a large amount of connections just fine. Do you think it would be worth ensuring that we only keep a single connection for consumers to use?